lib/supavisor/client_handler.ex

```diff
@@ -478,23 +544,34 @@ defmodule Supavisor.ClientHandler do
     end
   end
 
-  @spec db_checkout(:on_connect | :on_query, map()) :: pid() | nil
-  defp db_checkout(_, %{mode: :session, db_pid: db_pid}) when is_pid(db_pid) do
+  @spec db_checkout(:write | :read | nil, :on_connect | :on_query, map) :: {pid, pid} | nil
```

Comment:
Instead of `nil`, consider using `:both`, it may be a bit easier to find in case of errors.